sdk/src/types/block/semantic/mod.rs

```diff
@@ -105,17 +110,18 @@ impl<'a> SemanticValidationContext<'a> {
 
     ///
     pub fn validate(mut self) -> Result<Option<TransactionFailureReason>, Error> {
-        // Validation of inputs.
-        let mut has_implicit_account_creation_address = false;
-
-        self.commitment_context_input = self.transaction.context_inputs().commitment().copied();
+        self.commitment_context_input = self
+            .transaction
+            .context_inputs()
+            .commitment()
+            .map(|c| c.slot_commitment_id());
 
-        self.bic_context_input = self
+        self.bic_context_inputs = self
```

Comment:
same

Reply:
It really makes it much more easy to work with if we do the filter map here tbh

Reply:
Changed